internal/manager/controllerdef.go

```diff
@@ -42,50 +41,27 @@ func (c *ControllerDef) Name() string {
 	return reflect.TypeOf(c.Controller).String()
 }
 
-// MaybeSetupWithManager runs SetupWithManager on the controller if its EnablementStatus is either "enabled", or "auto"
-// and AutoHandler says that it should be enabled.
+// MaybeSetupWithManager runs SetupWithManager on the controller if it is enabled
+// and its AutoHandler (if any) indicates that it can load
 func (c *ControllerDef) MaybeSetupWithManager(mgr ctrl.Manager) error {
-	//nolint:exhaustive
-	switch *c.IsEnabled {
-	case util.EnablementStatusDisabled:
+	if !c.Enabled {
 		return nil
```

Comment:
consider adding a log entry like
```go
log.Info("controller disabled by a flag", "controller", c.Name())
```

and below
```go
log.Info("controller disabled by the auto-load handler", "controller", c.Name())
```

also consider logging the opposite case
```go
log.Info("controller enabled", "controller", c.Name())
```

Why? This may be useful for debugging if we ever need to see whether the logic evaluated to the controller starting up or not.

Reply:
This sounds like a good idea, but I am getting confused by the types here:

```
$ go vet ./...
# github.com/kong/kubernetes-ingress-controller/internal/manager
internal/manager/controllerdef.go:50:15: c.Controller.Log undefined (type Controller has no field or method Log)

$ git diff
diff --git a/internal/manager/controllerdef.go b/internal/manager/controllerdef.go
index 232b877a..7f90c35a 100644
--- a/internal/manager/controllerdef.go
+++ b/internal/manager/controllerdef.go
@@ -47,6 +47,7 @@ func (c *ControllerDef) Name() string {
 func (c *ControllerDef) MaybeSetupWithManager(mgr ctrl.Manager) error {
        switch *c.IsEnabled {
        case util.EnablementStatusDisabled:
+               c.Controller.Log.V4().Info("disabled by flag")
                return nil
        case util.EnablementStatusAuto:
                if c.AutoHandler == nil {
```
That sorta seems like it should work given that the Controller in a ControllerDef is a configuration.CoreV1WhateverReconciler that does have a `Log` struct inside it, but it doesn't since we just have the Controller interface here, which only guarantees the `SetupWithManager(ctrl.Manager)` method. Is there some way I can modify the interface such that it guarantees the Controller will have a Log object that implements some interface of its own?

Useful Golang education opportunity for me, but not something I know off the top of my head. If it's doable, we can add it later easily in ongoing work in #1585, so ignoring it for now.